java/src/main/native/src/NvtxRangeJni.cpp

```diff
@@ -39,11 +39,31 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_NvtxRange_push(JNIEnv *env, jclass cl
   CATCH_STD(env, );
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_NvtxRange_start(JNIEnv *env, jclass clazz, jstring name,
+                                                          jint color_bits) {
+  try {
+    cudf::jni::native_jstring range_name(env, name);
+    nvtx3::color range_color(static_cast<nvtx3::color::value_type>(color_bits));
+    nvtx3::event_attributes attr{range_color, range_name.get()};
+    auto nvtxRangeId = nvtxDomainRangeStartEx(nvtx3::domain::get<java_domain>(), attr.get());
```

Comment:
`nvtx3` has a `start_range` API: https://github.com/NVIDIA/NVTX/blob/3c98c8425b0376fd8653aac7cfc6a864f3897752/cpp/include/nvtx3/nvtx3.hpp#L2041

Reply:
We are using the `nvtx3.hpp` header here: `cudf/detail/nvtx/nvtx3.hpp`

Is this header going to be updated to the latest code in the repo you linked? Or should the NVTX3 api be brought in from that other repo?

This seems like a pretty simple API (the `nvtxRomainRangeStartEx` and `nvtxDomainRangeEnd`) so I am not sure we'll win a lot by using the C++ wrapper. Arguably we should pick nvtx3 or not, and `NvtxRangeJni` mixes them, so there's cleanup work we should do.

Reply:
> Is this header going to be updated to the latest code in the repo you linked? Or should the NVTX3 api be brought in from that other repo?

Naughty naught using things in `detail/` :) 

The right way to do it is have the JNI build pull in its own nvtx3 header from GitHub.

> This seems like a pretty simple API (the nvtxRomainRangeStartEx and nvtxDomainRangeEnd) so I am not sure we'll win a lot by using the C++ wrapper.

Yep, no real advantage, just wanted to point it out.